### PR TITLE
less: Restore original terminal state when SIGTERM is received

### PR DESCRIPTION
**Pull request by tcl3**
This PR makes 2 changes to less related to signal handling:

* The original terminal state is now restored when a `SIGTERM` is received, in the same way that it is when you exit the program normally.
* `SIGINT` is now ignored. This matches the behavior of `less` on Linux and FreeBSD.

Before:

https://github.com/SerenityOS/serenity/assets/2817754/0807e2f0-e21c-4974-8d76-6da7e80ab9c3

After:

https://github.com/SerenityOS/serenity/assets/2817754/fd66bf6b-3bf8-4f46-86bc-b29ee4c980f0

### CODE REVIEW

File: Userland/Utilities/less.cpp [hunk 1]
```diff
@@ -20,6 +20,8 @@ static struct termios g_save;
 // Flag set by a SIGWINCH signal handler to notify the main loop that the window has been resized.
 static Atomic<bool> g_resized { false };
 
+static Atomic<bool> g_should_switch_buffer { false };
```
Comment (petelliott) [inline Userland/Utilities/less.cpp:21]: can you call this something like `g_restore_buffer_on_close`. switch_buffer doesn't really convey when this variable will switch the buffer.

File: Userland/Utilities/less.cpp [hunk 2]
```diff
@@ -540,6 +542,18 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
+        teardown_tty();
+    });
+
+    auto teardown_sigaction_handler = [](auto) {
+        teardown_tty();
+        exit(1);
+    };
+    struct sigaction teardown_action;
+    teardown_action.sa_handler = teardown_sigaction_handler;
+    TRY(Core::System::sigaction(SIGTERM, &teardown_action, nullptr));
```
Comment (petelliott) [inline Userland/Utilities/less.cpp:553]: why are we mixing `sigaction` and `signal`. surely one of them is the better choice and we should stick to it?

File: Userland/Utilities/less.cpp [hunk 3]
```diff
@@ -181,8 +179,8 @@ class Pager {
     {
         // First, we get the current size of the window.
         struct winsize window;
-        if (ioctl(STDOUT_FILENO, TIOCGWINSZ, &window) == -1) {
-            perror("ioctl(2)");
+        if (auto maybe_error = Core::System::ioctl(STDOUT_FILENO, TIOCGWINSZ, &window); maybe_error.is_error()) {
+            warnln("ioctl(2): {}", strerror(maybe_error.error().code()));
```
Comment (petelliott) [inline Userland/Utilities/less.cpp:183]: not in the scope of this PR, but error should really have a way to get the error string, or strerror() if it's not present.

**Review by petelliott — CHANGES_REQUESTED** [review]

File: Userland/Utilities/less.cpp (hunk 2 above)
Comment (tcl3) [reply Userland/Utilities/less.cpp:555]: I didn't initially replace the existing `signal()` calls with `Core::System::sigaction()`, as I didn't think it was strictly necessary. However, after a [quick search](https://stackoverflow.com/a/232711) it seems that using `sigaction` is preferred, unless there is a good reason not to do so.

I've now replaced calls to `Core::System::signal()` with `Core::System::sigaction()`.

File: Userland/Utilities/less.cpp (hunk 3 above)
Comment (tcl3) [reply Userland/Utilities/less.cpp:183]: Agreed. The closest we can get to this at the moment is something like: `warnln("{}", maybe_error.release_error())`, but that introduces some formatting that we don't want here.  The above also changes behavior slightly, as it no longer prints to the debug console.

**Review by petelliott — APPROVED** [review]

**Pull request merged**